src/lazy/index.tsx

```diff
@@ -36,32 +38,22 @@ function lazyProxy<C extends ComponentType<any>>(
 
   LazyInternal.displayName = `Lazy(${displayNameFromId(moduleId)})`;
 
-  const getAssetUrls = (manifest: Manifest) => {
-    if (!manifest[moduleId]) {
+  /**
+   * Allows getting module chunks urls
+   */
+  const getAssetUrls = () => {
+    if (!SETTINGS.MANIFEST[moduleId]) {
       return;
     }
 
-    return manifest[moduleId];
+    return SETTINGS.MANIFEST[moduleId];
   };
 
   /**
-   * This will eventually be used to render preload link tags on transition.
-   * Currently not working as we need a way for the client to be able to know the manifest[moduleId].file
-   * without having to load the manifest on the client as it could be huge.
+   * Allows imperatively preload the module chunk asset
    */
   const preload = () => {
-    const head = document.querySelector('head');
-
-    if (!head) {
-      return;
-    }
-
-    const link = document.createElement('link');
-
-    link.rel = 'preload';
-
-    // TODO add href to link
-    head.appendChild(link);
+    preloadAsset(loader);
```

Comment:
Can we add `prefetch` too?

Reply:
I've opted to have `preload()` accept an optional `priority` argument. By default, it will figure out preload/prefetch based on the import type. But you can force high/low priority where `high = preload` and `low = prefetch`.
I've exported a `PRIORITY` const so we consumers don't have to think about numbers